src/components/ArrowMovingPlaceholder/index.test.js

```diff
@@ -0,0 +1,156 @@
+import ArrowPlaceholder from 'components/ArrowPlaceholder';
+import React from 'react';
+import SvgMousePosition from 'components/SvgMousePosition';
+import { shallow } from 'enzyme';
+import ArrowMovingPlaceholder from './component';
+
+const svg = document.createElement('svg');
+const shallowComponent = (props = {}) => shallow(<ArrowMovingPlaceholder {...props} />);
+
+describe('ArrowMovingPlaceholder Component', () => {
+  let eventsMap;
+  let defaultProps;
+
+  beforeEach(() => {
+    eventsMap = {};
+    svg.addEventListener = jest.fn((event, cb) => {
+      eventsMap[event] = cb;
+    });
+    svg.removeEventListener = jest.fn();
+
+    defaultProps = {
+      nodes: [
+        {
+          id: 'node-1',
+          states: [],
+          parents: [],
+          position: { x: 0, y: 0 },
+          size: { width: 160, height: 100 },
+        },
+        {
+          id: 'node-2',
+          states: [],
+          parents: [],
+          position: { x: 200, y: 200 },
+          size: { width: 160, height: 100 },
+        },
+      ],
+      node: {
+        id: 'node-1',
+        states: [],
+        parents: [],
+        position: { x: 0, y: 0 },
+        size: { width: 160, height: 100 },
+      },
+      fromPosition: { x: 50, y: 50 },
+      svg,
+      onSetPosition: jest.fn(),
+    };
+  });
+
+  describe('Event Listeners', () => {
+    describe('Check bind and unbind events', () => {
+      it('"mousedown" event', () => {
+        const component = shallowComponent(defaultProps);
+        const componentInstance = component.instance();
+
+        expect(svg.addEventListener)
+          .toHaveBeenCalledWith('mousedown', componentInstance.handleSetPosition);
+
+        component.unmount();
+
+        expect(svg.removeEventListener)
+          .toHaveBeenCalledWith('mousedown', componentInstance.handleSetPosition);
+      });
```

Comment:
I would go with something like that to be clearer about the description of the tests:

```
    describe('When binding mousedown', () => {
       it('calls handleSetPosition', () => {
         // expect 
       });
    });

    describe('When unbinding mousedown', () => {
       it('calls handleSetPosition', () => {
         // expect 
       });
     });
```

Reply:
Updated